app/src/main/java/com/arctouch/codechallenge/home/HomeFragment.java

```diff
@@ -0,0 +1,160 @@
+import com.arctouch.codechallenge.controller.HomeController;
+import com.arctouch.codechallenge.data.Cache;
+import com.arctouch.codechallenge.details.DetailsFragment;
+import com.arctouch.codechallenge.model.Movie;
+import com.arctouch.codechallenge.util.EndlessRecyclerViewScrollListener;
+
+import java.util.List;
+
+/**
+ * Created by wanderleyfilho on 7/22/18.
+ */
+
+public class HomeFragment extends Fragment {
+
+    private RecyclerView recyclerView;
+    private ProgressBar progressBar;
+    private HomeController homeController;
+    private long currentPage = 1L;
+    private List<Movie> movieList;
+    private EndlessRecyclerViewScrollListener endlessRecyclerViewScrollListener;
+    private boolean isSearching = false;
+    private String querySearch;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view =  inflater.inflate(R.layout.home_fragment, container, false);
+        setHasOptionsMenu(true);
+
+        this.recyclerView = view.findViewById(R.id.recyclerView);
+        this.progressBar = view.findViewById(R.id.progressBar);
+
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getActivity());
+        recyclerView.setLayoutManager(linearLayoutManager);
+        endlessRecyclerViewScrollListener = new EndlessRecyclerViewScrollListener(linearLayoutManager) {
+            @Override
+            public void onLoadMore(int page, int totalItemsCount, RecyclerView view) {
+                currentPage += 1L;
+                getNextMovies();
+            }
+        };
+        this.recyclerView.addOnScrollListener(endlessRecyclerViewScrollListener);
+
+        homeController = new HomeController();
+        homeController.loadGenres().subscribe(response -> {
+            Cache.setGenres(response.genres);
+            loadMovies();
+        });
+
+        return view;
+    }
+
+    public void loadMovies() {
+        homeController.loadUpcomingMoviesByPage(this.currentPage).subscribe(response -> {
+            movieList = homeController.addGenreToMovies(response.results);
+
+            recyclerView.setAdapter(new HomeAdapter(movieList, new HomeAdapter.OnItemClickListener(){
+                @Override
+                public void onItemClick(Movie movie) {
+                    DetailsFragment detailsFragment = new DetailsFragment();
+                    detailsFragment.movie = movie;
+                    getActivity().getSupportFragmentManager().beginTransaction().add(R.id.frame_layout,detailsFragment).addToBackStack(null).commit();
+                }
+            }));
+            progressBar.setVisibility(View.GONE);
+        });
+    }
+
+    public void getNextMovies() {
+        if (isSearching) {
+            homeController.searchMovies(this.currentPage, this.querySearch).subscribe(response -> {
+                movieList.addAll(homeController.addGenreToMovies(response.results));
```

Comment:
What would happen if your home controller calls emit an error like for instance, when your device doesn't have internet connection?

Reply:
Currently the application would crash, to solve it, the view should catch an exception from that call, in case it happens show a message on the screen instead of the movies list